samples/drivers/dac/sample.yaml

```diff
@@ -5,7 +5,8 @@ tests:
     tags: DAC
     platform_allow: |
       arduino_zero frdm_k22f frdm_k64f nucleo_f091rc nucleo_g071rb nucleo_g431rb
-      nucleo_l073rz nucleo_l152re twr_ke18f nucleo_f767zi nucleo_f429zi
+      nucleo_l073rz nucleo_l152re twr_ke18f nucleo_f767zi nucleo_f429zi bl652_dvk
+      bl653_dvk bl654_dvk
```

Comment:
This is beyond the scope of this PR, but @martinjaeger can you look at removing this list and just using the `depends_on: dac` below?

Reply:
Good point. Unfortunately, the supported resolution and channel number of the DAC cannot be obtained from devicetree, so we still have some `#ifdef`s in the sample code to determine it based on the selected board. I'll have a look if we can switch the `#ifdef`s to SOC-based versions and get rid of this growing allow-list. But that should not block this PR.